File: web/plugins/hex_to_number.js
import { WebPlugin } from "../engine/plugin.js";

export class HexToNumberPlugin extends WebPlugin {
    constructor() {
        super("16進数を数字にする");
    }

    async execute(line, context) {
        const match = line.match(/^16進数を数字にする\s+「(.+)」$/);
        if (!match) return;

        context.runtime.write(
            parseInt(match[1], 16)
        );
    }
}

File: pc/plugins/hex_to_number.py
import re
from core.plugin import Plugin


class HexToNumberPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'16進数を数字にする\s+「(.+)」', line)
        if not match:
            return None

        value = match.group(1)
        return f"print(int({value!r}, 16))"